### ISSUE TITLE
Build error with maturin 1.8.0

### ISSUE THREAD

**Issue by Staudey** [ijl/orjson#541]
Building with maturin 1.8.0 fails with the following error:

```
💥 maturin failed
  Caused by: `project.version` field is required in pyproject.toml unless it is present in the `project.dynamic` list
  ```
  
  Corresponding maturin change: https://github.com/PyO3/maturin/pull/2391
  
  > So rather than filling in the version, maturin should error and tell the user to add `dynamic = ["version"]` to their `pyproject.toml`
  (from https://github.com/PyO3/maturin/issues/2390)

**Comment by HythemK** [comment]
Met the same error yesterday

**Comment by esaaprillia** [comment]
the same mistake

**Comment by steals** [comment]
Same issue when trying to install orjson 3.10.12 on armv6

**Comment by HythemK** [comment]
> Same issue when trying to install orjson 3.10.12 on armv6

I soleved this by downloading tar.gz from pypi and then edit the pyproject.toml to right format.
Use `pip install .` to install it.

**Comment by ijl** [comment]
Fixed in [3.10.13](https://github.com/ijl/orjson/releases/tag/3.10.13). If you are not building strictly for your own use, there is nothing to do but upgrade to at least this version (or wait for maturin to possibly revert the breaking change).

**Comment by tobypeterson** [comment]
Now it doesn't build because it explicitly requires maturin 1.7.8. The PR from @Staudey should fix the issue in a more permanent fashion that won't require updating the version in pyproject.toml for each release.

**Comment by classabbyamp** [comment]
please fix this properly, like in #542. on Void Linux we can no longer build our `python3-orjson` package as we've already upgraded maturin to >=1.8.0.
